### PR TITLE
Robust lapper incremental

### PR DESCRIPTION
**Pull request by Topvennie**
Counts laps with only taking into account the new detections

### CODE REVIEW

File: .idea/git_toolbox_prj.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
```
Comment (redfast00) [inline .idea/git_toolbox_prj.xml:2]: Please add the `.idea` folder in your global .gitignore

File: src/main/java/telraam/logic/robustLapper/robustLapperIncremental/RobustLapperIncremental.java [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+
+        // Get the lapSourceId, create the source if needed
+        LapSourceDAO lapSourceDAO = jdbi.onDemand(LapSourceDAO.class);
+        lapSourceDAO.getByName(SOURCE_NAME).ifPresentOrElse(
+                lapSource -> this.lapSourceId = lapSource.getId(),
+                () -> this.lapSourceId = lapSourceDAO.insert(new LapSource(SOURCE_NAME))
+        );
+
+        setTeamVariables();
+        deleteLaps();
```
Comment (redfast00) [inline src/main/java/telraam/logic/robustLapper/robustLapperIncremental/RobustLapperIncremental.java:67]: If Telraam is restarted, I think all laps will be deleted, which is not desired

**Review by redfast00 — COMMENTED** [review]
Looks good :) some minor comments

File: .idea/git_toolbox_prj.xml (hunk 1 above)
Comment (Topvennie) [reply .idea/git_toolbox_prj.xml]: Commit a16dd42a463734703ec942a590344b35fc9fb3c4 got rid of it

File: src/main/java/telraam/logic/robustLapper/robustLapperIncremental/RobustLapperIncremental.java (hunk 2 above)
Comment (Topvennie) [reply src/main/java/telraam/logic/robustLapper/robustLapperIncremental/RobustLapperIncremental.java:67]: Now, when telraam gets restarted, all laps entered in the database by RobustLapperIncremental are deleted. The last detection is set "0", same for the last switchover. This means that with the next new detection it'll fetch everything from the database but after that it'll only fetch the new detections again. While this is not ideal, I still think it's fine as it only fetches everything once.

If you immediately want to work with only the new detections you’d have to delete the last two laps from each team. Look which team’s lap was set the earliest in timestamp and then get the detection that corresponds the closest in timestamp with that lap. Or you can just delete every lap entered and fetch all detections just once ;)

**Comment by FKD13** [comment]
I have run this on a simulation and the 2 lappers are performing quite differently.
Using this query:
```sql
select count(*), team_id, lap_source_id from lap where lap_source_id = 5 or lap_source_id = 6 group by team_id, lap_source_id order by team_id, lap_source_id;
```
I get the following result after about 10 minutes
scr
5 being the "old" robust lapper and 6 the new robust lapper. The old one is giving the expected results.

**Comment by FKD13** [comment]
Hmm might have been a endgecase when starting without many laps. When I restart telraam the counts match.

**Comment by FKD13** [comment]
After some time they definitely diverge
scr

**Pull request closed without merging**